src/App.tsx

```diff
@@ -15,48 +16,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType,
+  isReversed: boolean,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((item1, item2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return item1.length - item2.length;
+
+      case SortType.ALPHABET:
+        return item1.localeCompare(item2);
+
+      default:
+        return SortType.NONE;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  // eslint-disable-next-line no-console
+  console.log(sortType, isReversed);
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
```

Comment:
Add generic type if you use custom or non-primitive values.
```suggestion
  const [sortType, setSortType] = useState<SortType>(SortType.NONE);
```